raiden/transfer/token_network.py

```diff
@@ -0,0 +1,178 @@
+        del hashlocks_to_states[hashlock]
+
+
+def subdispatch_to_channel_by_id(token_network_state, state_change, block_number):
+    events = list()
+
+    ids_to_channels = token_network_state.channelidentifiers_to_channels
+    channel_state = ids_to_channels.get(state_change.channel_identifier)
+
+    if channel_state:
+        result = channel.state_transition(channel_state, state_change, block_number)
+        events.extend(result.events)
+
+    return TransitionResult(token_network_state, events)
+
+
+def handle_channel_close(token_network_state, state_change, block_number):
+    return subdispatch_to_channel_by_id(token_network_state, state_change, block_number)
+
+
+def handle_channelnew(token_network_state, state_change):
+    events = list()
+
+    channel_state = state_change.channel_state
+    channel_id = channel_state.identifier
+    our_address = channel_state.our_state.address
+    partner_address = channel_state.partner_state.address
+
+    token_network_state.network_graph.network.add_edge(
+        our_address,
+        partner_address,
+    )
+
+    token_network_state.channelidentifiers_to_channels[channel_id] = channel_state
+    token_network_state.partneraddresses_to_channels[partner_address] = channel_state
+
+    return TransitionResult(token_network_state, events)
+
+
+def handle_balance(token_network_state, state_change, block_number):
+    return subdispatch_to_channel_by_id(token_network_state, state_change, block_number)
+
+
+def handle_closed(token_network_state, state_change, block_number):
+    return subdispatch_to_channel_by_id(token_network_state, state_change, block_number)
+
+
+def handle_settled(token_network_state, state_change, block_number):
+    return subdispatch_to_channel_by_id(token_network_state, state_change, block_number)
+
+
+def handle_newroute(token_network_state, state_change):
+    events = list()
+
+    token_network_state.network_graph.network.add_edge(
+        state_change.participant1,
+        state_change.participant2,
+    )
+
+    return TransitionResult(token_network_state, events)
+
+
+def handle_action_transfer_direct(token_network_state, state_change, block_number):
+    receiver_address = state_change.receiver_address
+    channel_state = token_network_state.partneraddresses_to_channels.get(receiver_address)
+
+    if channel_state:
+        iteration = channel.state_transition(channel_state, state_change, block_number)
+        events = iteration.events
+    else:
+        failure = EventTransferSentFailed(
+            state_change.identifier,
+            'Unkown partner channel',
+        )
+        events = [failure]
+
+    return TransitionResult(token_network_state, events)
+
+
+def handle_receive_transfer_direct(token_network_state, state_change, block_number):
+    events = list()
+
+    channel_id = state_change.balance_proof.channel_address
+    channel_state = token_network_state.channelidentifiers_to_channels.get(channel_id)
+
+    if channel_state:
```

Comment:
So here and below if we receive transfers from non-partner nodes we ignore them here. Would logging something in DEBUG mode be prudent or overkill you think?

Reply:
well, if it's something of interest, I would add an event